src/Mvc/Mvc.RazorPages/src/Builder/RazorPagesEndpointRouteBuilderExtensions.cs

```diff
@@ -31,6 +33,253 @@ public static IEndpointConventionBuilder MapRazorPages(this IEndpointRouteBuilde
+        /// requests for dynamic content to a SPA framework, while also allowing requests for non-existent files to
+        /// result in an HTTP 404.
+        /// </para>
+        /// <para>
+        /// <see cref="MapFallbackToPage(IEndpointRouteBuilder, string)"/> registers an endpoint using the pattern
+        /// <c>{*path:nonfile}</c>. The order of the registered endpoint will be <c>int.MaxValue</c>.
+        /// </para>
+        /// <para>
+        /// <see cref="MapFallbackToPage(IEndpointRouteBuilder, string)"/> does not re-execute routing, and will
+        /// not generate route values based on routes defined elsewhere. When using this overload, the <c>path</c> route value
+        /// will be available. 
+        /// </para>
+        /// </remarks>
+        public static void MapFallbackToPage(this IEndpointRouteBuilder routes, string page)
+        {
+            if (routes == null)
+            {
+                throw new ArgumentNullException(nameof(routes));
+            }
+
+            if (page == null)
+            {
+                throw new ArgumentNullException(nameof(page));
+            }
+
+            PageConventionCollection.EnsureValidPageName(page, nameof(page));
+
+            EnsureRazorPagesServices(routes);
+
+            // Called for side-effect to make sure that the data source is registered.
+            GetOrCreateDataSource(routes);
+
+            // Maps a fallback endpoint with an empty delegate. This is OK because
+            // we don't expect the delegate to run. 
+            routes.MapFallback(context => Task.CompletedTask).Add(b =>
+            {
+                // MVC registers a policy that looks for this metadata.
+                b.Metadata.Add(CreateDynamicPageMetadata(page, area: null));
+            });
+        }
+
+        /// <summary>
+        /// Adds a specialized <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that will match
+        /// requests for non-file-names with the lowest possible priority. The request will be routed to a page endpoint that
+        /// matches <paramref name="page"/>.
+        /// </summary>
+        /// <param name="routes">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="page">The action name.</param>
+        /// <remarks>
+        /// <para>
+        /// <see cref="MapFallbackToPage(IEndpointRouteBuilder, string, string)"/> is intended to handle cases where URL path of
+        /// the request does not contain a file name, and no other endpoint has matched. This is convenient for routing
+        /// requests for dynamic content to a SPA framework, while also allowing requests for non-existent files to
+        /// result in an HTTP 404.
+        /// </para>
+        /// <para>
+        /// The order of the registered endpoint will be <c>int.MaxValue</c>.
```

Comment:
Where does this get set? I didn't find any references to the order being set in the code.

Reply:
`MapFallback` sets the order